libfwupdplugin/fu-common.c

```diff
@@ -2026,3 +2028,31 @@ fu_common_kernel_locked_down (void)
 	return FALSE;
 #endif
 }
+
+/**
+ * fu_common_is_cpu_intel:
+ *
+ * Uses CPUID to discover the CPU vendor and check if it is Intel.
+ *
+ * Return value: %TRUE if the vendor was Intel.
+ *
+ * Since: 1.5.0
+ **/
+gboolean
+fu_common_is_cpu_intel (void)
+{
+	guint eax = 0;
+	guint ebx = 0;
+	guint ecx = 0;
+	guint edx = 0;
+	guint level = 0;
+
+	/* get vendor */
+	__get_cpuid(level, &eax, &ebx, &ecx, &edx);
+	if (ebx == signature_INTEL_ebx &&
```

Comment:
I'm not familiar with this header and lookup but I notice you didn't also check eax. Is that intentional?

Reply:
I think eax is used to modify what gets returned in the other registers, i.e. it's a `in` parameter, but weirdly passed by reference.